### DISCUSSION TITLE
Add touch support for Drawer component to enable swipe gestures on mobile devices

### DISCUSSION THREAD

**Discussion by rtatarinov** [mantinedev/mantine#6715] [Feature requests]
I would like to request the implementation of touch gesture support for the Drawer component. Currently, the Drawer lacks the ability to respond to touch events, which limits its usability on mobile devices.

**Proposed Enhancement:**

To enhance the user experience, especially on touch devices, I propose adding a new prop that enables the Drawer to be hidden or revealed via swipe gestures. This feature should support the following:

1. **Swipe-to-Close**: The Drawer should close when the user swipes in the appropriate direction (e.g., swipe left on a right-aligned drawer).
2. **Swipe-to-Open** (optional): If feasible, the Drawer could also be opened via a swipe gesture, depending on the Drawer’s initial state and alignment.
3. **Animation**: Smooth animation support that matches the Drawer’s existing opening and closing transitions when triggered by a swipe.

**Example**:

For reference, you can see a similar implementation in the example provided here:
[Touch Gesture Drawer Example
](https://yastatic.net/s3/frontend/lego/storybook/index.html?path=/story/surface-drawer-touch-phone--playground)

**Comment by shagr4th** [comment]
FWIW someone did implement a swipeable mantine-based drawer here : https://github.com/AndrejNemec/mantine-vaul
But I would also prefer that swipe gestures are included in the core library

**Reply by fidalgodev** [reply]
Are there any plans for this? @rtivital

**Reply by rtivital** [reply]
Yes, it is planned to be implemented in 10.0 this year
